SvelteKit and adapter-node: npm run build fails on import

#### I am submitting a...

- [ ] Feature request
- [ ] Design defect
- [x] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

#### charts version:
@carbon/charts@0.41.92
@carbon/charts-svelte@0.41.92

#### Environment
node 16.5.0
@sveltejs/kit 1.0.0-next.137
@sveltejs/adapter-node 1.0.0-next.35
svelte 3.40.3
typescript 4.3.5

### Issue description
When I try to build my SvelteKit site with a component that uses @carbon/charts-svelte, I get the following error:

> > Using @sveltejs/adapter-node
> > Named export 'DonutChart' not found. The requested module '@carbon/charts' is a CommonJS module, which may not support all module.exports as named exports.
> CommonJS modules can always be imported via the default export, for example using:
> 
> import pkg from '@carbon/charts';
> const { DonutChart, LineChart } = pkg;
> 
> file:///Users/nates/dev/my-app/.svelte-kit/output/server/app.js:4
> import { DonutChart, LineChart } from "@carbon/charts";
>          ^^^^^^^^^^
> SyntaxError: Named export 'DonutChart' not found. The requested module '@carbon/charts' is a CommonJS module, which may not support all module.exports as named exports.
> CommonJS modules can always be imported via the default export, for example using:
> 
> import pkg from '@carbon/charts';
> const { DonutChart, LineChart } = pkg;
> 
>     at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
>     at async ModuleJob.run (node:internal/modules/esm/module_job:179:5)
>     at async Loader.import (node:internal/modules/esm/loader:178:24)
>     at async prerender (file:///Users/nates/dev/my-app/node_modules/@sveltejs/kit/dist/chunks/index6.js:80:14)
>     at async Object.prerender (file:///Users/nates/dev/my-app/node_modules/@sveltejs/kit/dist/chunks/index6.js:303:5)
>     at async adapt (file:///Users/nates/dev/my-app/node_modules/@sveltejs/adapter-node/index.js:82:4)
>     at async adapt (file:///Users/nates/dev/my-app/node_modules/@sveltejs/kit/dist/chunks/index6.js:329:2)
>     at async file:///Users/nates/dev/my-app/node_modules/@sveltejs/kit/dist/cli.js:907:5

I tried using the import recommended in the error...
```
import pkg from '@carbon/charts';
const { DonutChart, LineChart } = pkg;
```
but it threw an error...
> 'default' is not exported by node_modules/@carbon/charts/index.js, imported by src/routes/index.svelte

#### Steps to produce the issue
1. Install Node.js 16.5.0
2. `npm init svelte@next my-app` (choose SvelteKit demo app, TypeScript: Yes, eslint: Yes, and prettier: Yes)
3. `cd my-app`
3. `npm install --save-dev @sveltejs/adapter-node@1.0.0-next.35 @carbon/charts@0.41.92 @carbon/charts-svelte@0.41.92 d3@5.14.2 carbon-components@10.40.0`
4. Modify svelte.config.js to use the node-adapter and include @carbon/charts in optimizeDeps:
```typescript
import adapter from '@sveltejs/adapter-node'
import preprocess from 'svelte-preprocess';

/** @type {import('@sveltejs/kit').Config} */
const config = {
	preprocess: preprocess(),
	kit: {
		adapter: adapter({
			out: 'build'
		}),
		// hydrate the <div id="svelte"> element in src/app.html
		target: '#svelte',
		vite: {
			optimizeDeps: { include: ['@carbon/charts'] }
		}
	}
};

export default config;
```
5. Change /src/routes/index.svelte to:
```
<script lang="ts">
	import { DonutChart, LineChart } from '@carbon/charts-svelte'
</script>

<svelte:head>
	<title>Home</title>
	<link rel="stylesheet" href="https://unpkg.com/@carbon/charts/styles.min.css" />
</svelte:head>

<section>
	<DonutChart
	data={[
		{
			"group": "2V2N 9KYPM version 1",
			"value": 20000
		},
		{
			"group": "L22I P66EP L22I P66EP L22I P66EP",
			"value": 65000
		},
		{
			"group": "JQAI 2M4L1",
			"value": 75000
		},
		{
			"group": "J9DZ F37AP",
			"value": 1200
		},
		{
			"group": "YEL48 Q6XK YEL48",
			"value": 10000
		},
		{
			"group": "Misc",
			"value": 25000
		}
	]}
	options={{
		"title": "Donut (centered)",
		"resizable": true,
		"legend": {
			"alignment": "center"
		},
		"donut": {
			"center": {
				"label": "Browsers"
			},
			"alignment": "center"
		},
		"height": "400px"
	}}/>

<br/>
<LineChart
	data={[
	{
		"group": "Temperature",
		"date": "2019-01-01T05:00:00.000Z",
		"temp": 23
	},
	{
		"group": "Temperature",
		"date": "2019-02-01T05:00:00.000Z",
		"temp": 15
	},
	{
		"group": "Temperature",
		"date": "2019-03-01T05:00:00.000Z",
		"temp": 24
	},
	{
		"group": "Temperature",
		"date": "2019-04-01T04:00:00.000Z",
		"temp": 33
	},
	{
		"group": "Temperature",
		"date": "2019-05-01T04:00:00.000Z",
		"temp": 23
	},
	{
		"group": "Temperature",
		"date": "2019-06-01T04:00:00.000Z",
		"temp": 32
	},
	{
		"group": "Temperature",
		"date": "2019-07-01T04:00:00.000Z",
		"temp": 23
	},
	{
		"group": "Rainfall",
		"date": "2019-01-01T05:00:00.000Z",
		"rainfall": 50
	},
	{
		"group": "Rainfall",
		"date": "2019-02-01T05:00:00.000Z",
		"rainfall": 65
	},
	{
		"group": "Rainfall",
		"date": "2019-03-01T05:00:00.000Z",
		"rainfall": 35
	},
	{
		"group": "Rainfall",
		"date": "2019-04-01T04:00:00.000Z",
		"rainfall": 43
	},
	{
		"group": "Rainfall",
		"date": "2019-05-01T04:00:00.000Z",
		"rainfall": 53
	},
	{
		"group": "Rainfall",
		"date": "2019-06-01T04:00:00.000Z",
		"rainfall": 19
	},
	{
		"group": "Rainfall",
		"date": "2019-07-01T04:00:00.000Z",
		"rainfall": 13
	}
]}
options={{
	"title": "Line + Line (dual axes)",
	"axes": {
		"left": {
			"title": "Temperature (°C)",
			"mapsTo": "temp"
		},
		"bottom": {
			"scaleType": "time",
			"mapsTo": "date",
			"title": "Date"
		},
		"right": {
			"title": "Rainfall (mm)",
			"mapsTo": "rainfall",
			"correspondingDatasets": [
				"Rainfall"
			]
		}
	},
	"curve": "curveMonotoneX",
	"height": "400px"
}}
/>
</section>
```
 6. Run `npm run build`
 
#### Current behavior
Error displays

#### Expected behavior
No error

### Screenshot or recording

After adding the kit.vite.ssr.noExternal property, the error changed to:

> > Using @sveltejs/adapter-node
> > Cannot find module '/Users/nates/dev/my-app/node_modules/carbon-components/es/globals/js/settings' imported from /Users/nates/dev/my-app/.svelte-kit/output/server/app.js
> Did you mean to import carbon-components/es/globals/js/settings.js?
> Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/nates/dev/my-app/node_modules/carbon-components/es/globals/js/settings' imported from /Users/nates/dev/my-app/.svelte-kit/output/server/app.js
> Did you mean to import carbon-components/es/globals/js/settings.js?
>     at new NodeError (node:internal/errors:370:5)
>     at finalizeResolution (node:internal/modules/esm/resolve:321:11)
>     at moduleResolve (node:internal/modules/esm/resolve:756:10)
>     at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:867:11)
>     at Loader.resolve (node:internal/modules/esm/loader:89:40)
>     at Loader.getModuleJob (node:internal/modules/esm/loader:242:28)
>     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
>     at link (node:internal/modules/esm/module_job:75:36)
> 

This can reproduced using the SvelteKit demo with the steps above. I also updated all the dependencies to the latest version.

--- a/packages/svelte/README.md
+++ b/packages/svelte/README.md
@@ -291,12 +291,12 @@ that fires when hovering over a bar.
   }
 
   onMount(() => {
-    chart.services.events.addEventListener("bar-mouseover", barMouseOver);
-
     return () => {
-      chart.services.events.removeEventListener("bar-mouseover", barMouseOver);
+      if (chart) chart.services.events.removeEventListener("bar-mouseover", barMouseOver);
     };
   });
+
+  $: if (chart) chart.services.events.addEventListener("bar-mouseover", barMouseOver);
 </script>
 
 <BarChartSimple

--- a/packages/svelte/README.md
+++ b/packages/svelte/README.md
@@ -35,12 +35,12 @@ bundled version of the library.
 
 This is an overview of using Carbon Charts with common Svelte set-ups.
 
--   [sveltekit](#sveltekit)
--   [vite](#vite)
--   [sapper](#sapper)
--   [rollup](#rollup)
--   [webpack](#webpack)
--   [snowpack](#snowpack)
+-   [SvelteKit](#sveltekit)
+-   [Vite](#vite)
+-   [Sapper](#sapper)
+-   [Rollup](#rollup)
+-   [Webpack](#webpack)
+-   [Snowpack](#snowpack)
 
 ### SvelteKit
 
@@ -51,13 +51,27 @@ framework for building Svelte apps that supports both client-side rendering
 For set-ups powered by [vite](https://github.com/vitejs/vite), add
 `@carbon/charts` to the list of dependencies for `vite` to optimize.
 
+If using a [SvelteKit adapter](https://kit.svelte.dev/docs#adapters), instruct
+`vite` to avoid externalizing `@carbon/charts` when building for production.
+
 ```js
 // svelte.config.js
+import adapter from '@sveltejs/adapter-node';
+
+const production = process.env.NODE_ENV === 'production';
+
+/** @type {import('@sveltejs/kit').Config} */
 const config = {
 	kit: {
+		adapter: adapter(),
 		target: '#svelte',
 		vite: {
-			optimizeDeps: { include: ['@carbon/charts'] },
+			optimizeDeps: {
+				include: ['@carbon/charts'],
+			},
+			ssr: {
+				noExternal: [production && '@carbon/charts'].filter(Boolean),
+			},
 		},
 	},
 };

